cmake/MacApp.cmake

```diff
@@ -0,0 +1,58 @@
+# This is Mac-only.
+if (NOT APPLE)
+    RETURN()
+endif (NOT APPLE)
+
+# The source tree containing certain macOS resources.
+SET(OSX_DIR ${CMAKE_CURRENT_SOURCE_DIR}/osx)
+
+SET(RESOURCE_FILES
+    ${OSX_DIR}/launch_fish.scpt
+    ${OSX_DIR}/fish_term_icon.icns
+    ${CMAKE_CURRENT_SOURCE_DIR}/build_tools/osx_package_scripts/add-shell
+    ${OSX_DIR}/install.sh
+)
+
+# Resource files must be present in the source list.
+ADD_EXECUTABLE(fish_macapp EXCLUDE_FROM_ALL
+    ${OSX_DIR}/osx_fish_launcher.m
+    ${RESOURCE_FILES}
+)
+
+# Compute the version. Note this is done at generation time, not build time,
+# so cmake must be re-run after version changes for the app to be updated. But
+# generally this will be run by make_pkg.sh which always re-runs cmake.
+EXECUTE_PROCESS(
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/build_tools/git_version_gen.sh --stdout
```

Comment:
It might be made to work, but would require a lot of custom code to modify the Info.plist. In practice this app is only built at release time so I don't think it will be a problem; if it is we can add it later.